pkg/extensions/sync/service.go

```diff
@@ -348,164 +315,321 @@ func (service *BaseService) SyncRepo(ctx context.Context, repo string) error {
+	// var shouldCommit bool
 	remoteImageRef, err := service.remote.GetImageReference(remoteRepo, tag)
 	if err != nil {
 		service.log.Error().Err(err).Str("errortype", common.TypeOf(err)).
 			Str("repository", remoteRepo).Str("reference", tag).Msg("couldn't get a remote image reference")
 
-		return "", err
+		return err
 	}
 
-	_, mediaType, manifestDigest, err := service.remote.GetManifestContent(remoteImageRef)
+	defer service.rc.Close(ctx, remoteImageRef)
+
+	_, remoteManifestDesc, isConverted, err := service.remote.GetOCIManifest(ctx, remoteRepo, tag)
 	if err != nil {
 		service.log.Error().Err(err).Str("repository", remoteRepo).Str("reference", tag).
-			Msg("couldn't get upstream image manifest details")
+			Msg("failed to get upstream image manifest details")
 
-		return "", err
+		return err
 	}
 
-	if !isSupportedMediaType(mediaType) {
-		return "", zerr.ErrMediaTypeNotSupported
+	referrers, err := service.rc.ReferrerList(ctx, remoteImageRef)
+	if err != nil {
+		return err
 	}
 
 	if service.config.OnlySigned != nil && *service.config.OnlySigned &&
-		!references.IsCosignTag(tag) && !common.IsReferrersTag(tag) {
-		signed := service.references.IsSigned(ctx, remoteRepo, manifestDigest.String())
+		!isCosignTag(tag) && !common.IsReferrersTag(tag) {
+		signed := hasSignatureReferrers(referrers)
 		if !signed {
 			// skip unsigned images
-			service.log.Info().Str("image", remoteImageRef.DockerReference().String()).
+			service.log.Info().Str("image", remoteImageRef.CommonName()).
 				Msg("skipping image without mandatory signature")
 
-			return "", zerr.ErrSyncImageNotSigned
+			return zerr.ErrSyncImageNotSigned
 		}
 	}
 
-	skipImage, err := service.destination.CanSkipImage(destinationRepo, tag, manifestDigest)
+	localImageRef, err := service.destination.GetImageReference(localRepo, tag)
 	if err != nil {
 		service.log.Error().Err(err).Str("errortype", common.TypeOf(err)).
-			Str("repository", destinationRepo).Str("reference", tag).
-			Msg("couldn't check if the local image can be skipped")
+			Str("repo", localRepo).Str("reference", localImageRef.Tag).Msg("failed to get a local image reference")
+
+		return err
 	}
 
-	if !skipImage {
-		localImageRef, err := service.destination.GetImageReference(destinationRepo, tag)
-		if err != nil {
-			service.log.Error().Err(err).Str("errortype", common.TypeOf(err)).
-				Str("repository", destinationRepo).Str("reference", tag).Msg("couldn't get a local image reference")
+	defer service.rc.Close(ctx, localImageRef)
 
-			return "", err
-		}
+	// first sync image
+	_, err = service.syncReference(ctx, localRepo, remoteImageRef, localImageRef, remoteManifestDesc.Digest, false)
+	if err != nil {
+		return err
+	}
 
-		service.log.Info().Str("remote image", remoteImageRef.DockerReference().String()).
-			Str("local image", fmt.Sprintf("%s:%s", destinationRepo, tag)).Msg("syncing image")
+	// shouldCommit = !skipped
```

Comment:
Commented code to be cleaned up. There's also other places where this needs to be done.